src/morphology/english/getVerbForms.js

```diff
@@ -0,0 +1,217 @@
+		};
+	}
+
+	if ( verbPrefixesRegex.threeLetterPrefixes.test( word ) === true ) {
+		return {
+			normalizedWord: word.replace( verbPrefixesRegex.threeLetterPrefixes, "" ),
+			prefix: word.substring( 0, 3 ),
+		};
+	}
+
+	if ( verbPrefixesRegex.twoLetterHyphenPrefixes.test( word ) === true ) {
+		return {
+			normalizedWord: word.replace( verbPrefixesRegex.twoLetterHyphenPrefixes, "" ),
+			prefix: word.substring( 0, 3 ),
+		};
+	}
+
+	if ( verbPrefixesRegex.twoLetterPrefixes.test( word ) === true ) {
+		return {
+			normalizedWord: word.replace( verbPrefixesRegex.twoLetterPrefixes, "" ),
+			prefix: word.substring( 0, 2 ),
+		};
+	}
+
+	if ( verbPrefixesRegex.oneLetterPrefixes.test( word ) === true ) {
+		return {
+			normalizedWord: word.replace( verbPrefixesRegex.oneLetterPrefixes, "" ),
+			prefix: word.substring( 0, 1 ),
+		};
+	}
+};
+
+
+const checkIrregulars = function( word ) {
+	let irregulars;
+
+	irregularVerbs.forEach( function( paradigm ) {
+		paradigm.forEach( function( wordInParadigm ) {
+			if ( wordInParadigm === word ) {
+				irregulars = paradigm;
+			}
+		} );
+	} );
+
+	if ( isUndefined( irregulars ) ) {
+		const normalizedIrregular = normalizePrefixed( word );
+
+		if ( ! isUndefined( normalizedIrregular ) ) {
+			irregularVerbs.forEach( function( paradigm ) {
+				paradigm.forEach( function( wordInParadigm ) {
+					if ( wordInParadigm === normalizedIrregular.normalizedWord ) {
+						irregulars = paradigm.map( function( verb ) {
+							return normalizedIrregular.prefix.concat( verb );
+						} );
+					}
+				} );
+			} );
+		}
+	}
+
+	return irregulars;
+};
+
+const endsWithS = function( word ) {
+	return word[ word.length - 1 ] === "s";
+};
+
+const endsWithIng = function( word ) {
+	return word.substring( word.length - 3, word.length ) === "ing";
+};
+
+const endsWithEd = function( word ) {
+	return word.substring( word.length - 2, word.length ) === "ed";
+};
+
+const buildVerbFormFromRegex = function( word, regex ) {
+	for ( let i = 0; i < regex.length; i++ ) {
+		if ( regex[ i ].reg.test( word ) === true ) {
+			return word.replace( regex[ i ].reg, regex[ i ].repl );
+		}
+	}
+};
+
+const sFormToInfinitive = function( word ) {
+	return buildVerbFormFromRegex( word, sFormToInfinitiveRegex );
+};
+
+const ingFormToInfinitive = function( word ) {
+	return buildVerbFormFromRegex( word, ingFormToInfinitiveRegex );
+};
+
+const edFormToInfinitive = function( word ) {
+	return buildVerbFormFromRegex( word, edFormToInfinitiveRegex );
+};
+
+const infinitiveToSForm = function( word ) {
+	return buildVerbFormFromRegex( word, infinitiveToSFormRegex );
+};
+
+const infinitiveToIngForm = function( word ) {
+	return buildVerbFormFromRegex( word, infinitiveToIngFormRegex );
+};
+
+const infinitiveToEdForm = function( word ) {
+	return buildVerbFormFromRegex( word, infinitiveToEdFormRegex );
+};
+
+const getInfinitive = function( word ) {
+	let infinitive = word;
+	let guessedForm = "inf";
+
+	if ( endsWithS( word ) ) {
```

Comment:
Say that `endsWithS` returns true and the infinitive/guessedForm are set accordingly, the function would still continue to check whether `endsWithIng` and `endsWithEd` are also true. But that's not really necessary because only one of these conditions can hold true, right? That's why I think it would be better to do an early return right when an if-condition evaluates to `true`.

Reply:
Fixed.